test/SecTester.Scan.Tests/ScanSettingsTests.cs

```diff
@@ -0,0 +1,237 @@
+    _targetOptions.Headers.Returns(new List<KeyValuePair<string, IEnumerable<string>>>());
+    _targetOptions.Method.Returns(HttpMethod.Get);
+    _targetOptions.Query.Returns(new List<KeyValuePair<string, string>>());
+
+    _scanSettingsOptions.Name.Returns(ScanName);
+    _scanSettingsOptions.Smart.Returns(false);
+    _scanSettingsOptions.Target.Returns(_targetOptions);
+    _scanSettingsOptions.RepeaterId.Returns(RepeaterId);
+    _scanSettingsOptions.PoolSize.Returns(1);
+    _scanSettingsOptions.TargetTimeout.Returns(TimeSpan.FromSeconds(5));
+    _scanSettingsOptions.SlowEpTimeout.Returns(TimeSpan.FromSeconds(200));
+    _scanSettingsOptions.SkipStaticParams.Returns(false);
+    _scanSettingsOptions.Tests.Returns(new List<TestType> { TestType.Csrf });
+    _scanSettingsOptions.AttackParamLocations.Returns(new List<AttackParamLocation> { AttackParamLocation.Query });
+  }
+
+  public void Dispose()
+  {
+    _targetOptions.ClearSubstitute();
+    _scanSettingsOptions.ClearSubstitute();
+  }
+
+  [Fact]
+  public void PublicConstructor_CreatesInstanceWithDefaultOptions()
+  {
+    // act
+    var result = new ScanSettings(_targetOptions, new List<TestType> { TestType.Csrf });
+
+    // assert
+    result.Should().BeEquivalentTo(new
+    {
+      RepeaterId = null as string,
+      Name = "GET example.com",
+      PoolSize = 10,
+      TargetTimeout = TimeSpan.FromSeconds(5),
+      SlowEpTimeout = TimeSpan.FromSeconds(1000),
+      Smart = true,
+      SkipStaticParams = true,
+      Tests = new List<TestType> { TestType.Csrf },
+      AttackParamLocations = new List<AttackParamLocation>
+      {
+        AttackParamLocation.Body, AttackParamLocation.Query, AttackParamLocation.Fragment
+      },
+      Target = new { Url = TargetUrl }
+    });
+  }
+
+  [Fact]
+  public void InternalConstructor_CreatesInstance()
+  {
+    // act
+    var result = new ScanSettings(_scanSettingsOptions);
+
+    // assert
+    result.Should().BeEquivalentTo(new
+    {
+      RepeaterId,
+      Name = ScanName,
+      PoolSize = 1,
+      TargetTimeout = TimeSpan.FromSeconds(5),
+      SlowEpTimeout = TimeSpan.FromSeconds(200),
+      Smart = false,
+      SkipStaticParams = false,
+      Tests = new List<TestType> { TestType.Csrf },
+      AttackParamLocations = new List<AttackParamLocation> { AttackParamLocation.Query },
+      Target = new { Url = TargetUrl }
+    });
+  }
+
+  [Fact]
+  public void InternalConstructor_NameIsNull_CreatesInstanceWithDefaultName()
+  {
+    // arrange
+    _scanSettingsOptions.Name.ReturnsForAnyArgs(_ => null!);
+
+    // act
+    var result = new ScanSettings(_scanSettingsOptions);
+
+    // assert
+    result.Should().BeEquivalentTo(new { Name = "GET example.com" });
+  }
+
+  [Fact]
+  public void InternalConstructor_NameIsNull_CreatesInstanceTruncatedHost()
+  {
+    // arrange
+    _scanSettingsOptions.Name.ReturnsForAnyArgs(_ => null);
```

Comment:
Why do you use `ReturnsForAnyArgs` if this is a property that definitely does not accept any arguments?